tactics/tacticals.ml

```diff
@@ -367,9 +367,36 @@ module New = struct
     tclTHENSFIRSTn t1 l (tclUNIT())
   let tclTHENFIRST t1 t2 =
     tclTHENFIRSTn t1 [|t2|]
+
+  let tclBINDFIRST t1 t2 =
+    t1 >>= fun ans ->
+    Proofview.Unsafe.tclGETGOALS >>= fun gls ->
+    match gls with
+    | [] -> tclFAIL 0 (str "Expect at least one goal.")
+    | hd::tl ->
+    Proofview.Unsafe.tclSETGOALS [List.hd gls] <*> t2 ans >>= fun ans ->
+    Proofview.Unsafe.tclNEWGOALS (List.tl gls) <*>
+    Proofview.tclUNIT ans
```

Comment:
Please use `hd` and `tl` instead of using `List.hd` and `List.tl`, that gives stronger static invariants.

Reply:
Indeed. A stupid overlook. Changed (and rebased).